src/src/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -559,12 +521,15 @@ private boolean verifyAccount(Account[] accountList, String username, String uni
         return true;
     }
 
-    private boolean hasSupportToAddUserThroughBroker() {
-        Intent intent = new Intent();
-        intent.setPackage(AuthenticationConstants.Broker.AZURE_AUTHENTICATOR_APP_PACKAGE_NAME);
-        intent.setClassName(AuthenticationConstants.Broker.AZURE_AUTHENTICATOR_APP_PACKAGE_NAME,
-                AuthenticationConstants.Broker.AZURE_AUTHENTICATOR_APP_PACKAGE_NAME
-                        + ".ui.AccountChooserActivity");
+    /**
+     * True if broker has multi-user support. ADAL is checking the existence of
+     * .ui.AccountChooserActivity which only exists in the new broker with PRT
+     * support.
+     */
+    private boolean hasSupportToAddUserThroughBroker(final String brokerPackageName) {
+        final Intent intent = new Intent();
+        intent.setPackage(brokerPackageName);
+        intent.setClassName(brokerPackageName, brokerPackageName + ".ui.AccountChooserActivity");
```

Comment:
It's usually bad practice to hardcode class names, there are helper methods such as queryIntentActivities to detect activity that can handle intent...